src/controllers/Game.ts

```diff
@@ -154,20 +131,33 @@ export default class Game {
     }
 
 
-    async getScoresaber(playerId: string): Promise<any> {
+    async getScoresaber(playerId: string): Promise<ScoreSaberPlayerInfo | null> {
         try {
-            const response = await fetch(`https://scoresaber.com/api/player/${playerId}/full`);
-            if (response.status === 404) {
-                return {};
+            const response = await got.get(`https://scoresaber.com/api/player/${playerId}/full`);
+            switch (response.statusCode) {
+                case 200:
+                    return JSON.parse(response.body) as ScoreSaberPlayerInfo;
+                case 404:
+                    return null;
+                default:
+                    return null;
             }
-
-            return await response.json();
         } catch (e) {
-            console.error(e);
-            await new Promise((resolve) => setTimeout(resolve, 1000));
-            return await this.getScoresaber(playerId);
+            logger.warn(`Failed to get scoresaber info for ${playerId}, retrying...`);
+
+            let attempts = 0;
+            while (attempts < 3) {
+                attempts++;
+                try {
+                    return await this.getScoresaber(playerId);
+                } catch (e) {
+                    logger.warn(`Failed to get scoresaber info for player ${playerId} on attempt ${attempts}/3: ${e as string}`);
```

Comment:
`e.toString()` pls

Reply:
An early eslint configuration was angry, it should be chill now